libraries/CAN_comms/Inc/CAN_comms.h

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file CAN_comms.h
+ * 
+ * @brief CAN communications layer library header.
+ * This header contains user facing functions declarations and structs for the CAN communications layer.
+ * 
+ * @attention See README for details on how to implement this library.
+ */
+
+#ifndef CAN_COMMS_H
+#define CAN_COMMS_H
+
+#include <stdint.h>
+
+#include "can.h"
```

Comment:
Is including can.h necessary? Or can we remove it? I wanted to declare the callback inside of can.h but cant because CAN comms.h will include can.h meaning it will try to find the defintion a typedef you have defined in CAN_comms.h.

Reply:
CAN_comms requires some declarations and typedefs that are included in can.h. However, I agree that defining the Rx callback in can.h makes the most sense. Let me do some digging to see if there is a way to resolve this.

Reply:
Removed can.h and it works